repldex/discordbot/bot.py

```diff
@@ -240,8 +238,8 @@ async def on_ready():
 	await client.change_presence(activity=discord.Game(name=prefix + 'help'))
 
 
-def discord_id_to_user(user_id):
-	user = client.get_user(user_id)
+async def discord_id_to_user(user_id):
+	user = client.get_user(user_id) or await client.fetch_user(user_id)
 	return str(user) if user else None
```

Comment:
Users and members should be cached in two dicts (for users and members) with the user id as the key, because otherwise we're spamming the API. This should be changed everywhere we do fetch_user or fetch_member, maybe create utility functions that do the same as fetch_(user|member) but with caching?

Reply:
aight lemme see if i can implement a cache